File: ALODWebUtility/Printing/FormFieldParsers/NonHierarchicalParser.vb
Imports ALOD.Core.Interfaces
Imports WebSupergoo.ABCpdf8
Imports WebSupergoo.ABCpdf8.Objects

Namespace Printing.FormFieldParsers

    Public Class NonHierarchicalParser
        Implements IPDFFormFieldParser

        Public Function ParseField(ByVal document As Doc, ByVal desiredField As String) As Field Implements IPDFFormFieldParser.ParseField
            Dim fieldNames As String() = document.Form.GetFieldNames()
            Dim fullFieldName As String = fieldNames.First(Function(x) x.Equals(desiredField.Trim()))
            Return document.Form(fullFieldName)
        End Function

    End Class

End Namespace

File: ALODWebUtility.CS/Printing/FormFieldParsers/NonHierarchicalParser.cs
using System.Linq;
using ALOD.Core.Interfaces;
using WebSupergoo.ABCpdf8;
using WebSupergoo.ABCpdf8.Objects;

namespace ALODWebUtility.Printing.FormFieldParsers
{
    public class NonHierarchicalParser : IPDFFormFieldParser
    {
        public Field ParseField(Doc document, string desiredField)
        {
            string[] fieldNames = document.Form.GetFieldNames();
            string fullFieldName = fieldNames.First(x => x.Equals(desiredField.Trim()));
            return document.Form[fullFieldName];
        }
    }
}
